src/include/compat.h

```diff
@@ -145,8 +145,10 @@
     #define ceph_pthread_setname pthread_setname_np
   #endif
 #elif defined(HAVE_PTHREAD_SET_NAME_NP)
-  /* Fix a small name diff */
-  #define ceph_pthread_setname pthread_set_name_np
+  /* Fix a small name diff and return 0 */
+  #define ceph_pthread_setname(thread, name) ({ \
+    pthread_set_name_np(thread, name);          \
+    0; })
```

Comment:
no need to do this on GNU/Linux.

Reply:
@tchaikov 
`pthread_set_name_np` is only found on FreeBSD.
Si I'd expect this to be only selected on FreeBSD.

Reply:
https://git.musl-libc.org/cgit/musl/commit/?id=8fb28b0b3e7a5e958fb844722a4b2ef9bc244af1